Album/ViewControllers/DetailsViewController.m

```diff
@@ -6,34 +6,133 @@
+    [likeQuery findObjectsInBackgroundWithBlock:^(NSArray *statuses, NSError *error) {
+                   if (statuses != nil) {
+                   NSLog(@"Successfully fetched like statuses!");
+                   // If no like status, create one
+                   if (statuses.count == 0) {
+                   self.likeStatus = [[UserPin alloc] init];
+                   self.likeStatus.userId = self.currentUser.objectId;
+                   self.likeStatus.pinId = self.pin.objectId;
+                   self.likeStatus.hasLiked = NO;
+                   [self.likeStatus saveInBackground];
+                   [self.likeButton setImage:[UIImage systemImageNamed:@"heart"] forState:UIControlStateNormal];
+                   } else {
+                   // Set like button corresponding to the hasLiked field
+                   self.likeStatus = statuses[0];
+                   if (self.likeStatus.hasLiked == YES) {
+                    [self.likeButton setImage:[UIImage systemImageNamed:@"heart.fill"] forState:UIControlStateNormal];
+                   } else {
+                    [self.likeButton setImage:[UIImage systemImageNamed:@"heart"] forState:UIControlStateNormal];
+                   }
+                   }
+                   NSString *formattedString = [NSString stringWithFormat:@"%@ likes", self.pin.likeCount];
+                   [self.likeButton setTitle:formattedString forState:UIControlStateNormal];
+                   } else {
+                   NSLog(@"%@", error.localizedDescription);
+                   }
+               }];
+} /* setLikeStatus */
+
+- (void)tapLiked:(UITapGestureRecognizer *)sender {
+    if (sender.state == UIGestureRecognizerStateRecognized) {
+        // Update like status of pin
+        self.likeStatus.hasLiked = !self.likeStatus.hasLiked;
+        // Update count and button view accordingly
+        if (self.likeStatus.hasLiked == YES) {
+            self.pin.likeCount = @([self.pin.likeCount intValue] + 1);
+            if ([self.pin.likeCount intValue] < 0) {
+                self.pin.likeCount = @(0);
+            }
+            [self.likeButton setImage:[UIImage systemImageNamed:@"heart.fill"] forState:UIControlStateNormal];
+        } else {
+            self.pin.likeCount = @([self.pin.likeCount intValue] - 1);
+            if ([self.pin.likeCount intValue] < 0) {
+                self.pin.likeCount = @(0);
+            }
+            [self.likeButton setImage:[UIImage systemImageNamed:@"heart"] forState:UIControlStateNormal];
+        }
+        NSString *formattedString = [NSString stringWithFormat:@"%@ likes", self.pin.likeCount];
+        [self.likeButton setTitle:formattedString forState:UIControlStateNormal];
+        [self.pin saveInBackground];
+        // Save the status object in background
+        [self.likeStatus saveInBackground];
+    }
+} /* tapLiked */
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
```

Comment:
For methods related to protocol conformance preferably use pragma marks, fe `#pragma mark - <UICollectionViewDataSource` or `#pragma mark - UICollectionViewDelegate` it'll help keep your code better orginized, [here](https://nshipster.com/pragma/) a small post :)

Reply:
Refactored all files!